packages-ts/gauntlet-terra-contracts/src/commands/abstract/executionWrapper.ts

```diff
@@ -13,46 +21,68 @@ export interface AbstractInstruction<Input, ContractInput> {
   makeInput: (flags: any, args: string[]) => Promise<Input>
   validateInput: (input: Input) => boolean
   makeContractInput: (input: Input) => Promise<ContractInput>
+  beforeExecute?: (context: BeforeExecutionContext<Input, ContractInput>) => () => Promise<void>
   afterExecute?: (response: Result<TransactionResponse>) => any
 }
 
-export const instructionToCommand = (instruction: AbstractInstruction<any, any>) => {
+export const instructionToCommand = <Input, ContractInput>(instruction: AbstractInstruction<Input, ContractInput>) => {
   const id = `${instruction.instruction.contract}:${instruction.instruction.function}`
   const category = `${instruction.instruction.category}`
   const examples = instruction.examples || []
   return class Command extends TerraCommand {
     static id = id
     static category = category
     static examples = examples
+
     command: AbstractCommand
 
     constructor(flags, args) {
       super(flags, args)
     }
 
-    afterExecute = instruction.afterExecute
+    defaultBeforeExecute = (context: BeforeExecutionContext<Input, ContractInput>) => async () => {
```

Comment:
Imo `defaultBeforeExecute` doesn't belong here as a class member, I'd move it out to `instructionToCommand` or the whole file context

Reply:
Right, will do